File: Practical6.py
''' Practical 6:-
Expert System for Hospitals and Medical Facilities '''

print("========================================")
print(" Hospital Expert System ")
print("========================================")

# Knowledge base using dictionary
disease_data = {
    "fever": {
        "disease": "Flu",
        "advice": "Take rest and drink plenty of fluids."
    },

    "cough": {
        "disease": "Cold",
        "advice": "Use cough syrup and stay hydrated."
    },

    "headache": {
        "disease": "Migraine",
        "advice": "Take proper sleep and avoid stress."
    },

    "stomach pain": {
        "disease": "Acidity",
        "advice": "Avoid oily food and drink more water."
    },

    "chest pain": {
        "disease": "Heart Problem",
        "advice": "Consult a doctor immediately."
    }
}

print("\nAvailable symptoms:")
for symptom in disease_data:
    print("-", symptom)

# User input
user_symptom = input("\nEnter your symptom: ").lower()

# Expert system logic
if user_symptom in disease_data:

    print("\nPossible Disease:",
          disease_data[user_symptom]["disease"])

    print("Medical Advice:",
          disease_data[user_symptom]["advice"])

else:
    print("\nSorry! Symptom not found in database.")
    print("Please consult a doctor.")


'''
========================================
 Hospital Expert System
========================================

Available symptoms:
- fever
- cough
- headache
- stomach pain
- chest pain

Enter your symptom: fever

Possible Disease: Flu
Medical Advice: Take rest and drink plenty of fluids.
'''

File: Practical6.cpp
/*
Practical 6:
Expert System for Hospitals and Medical Facilities
*/
/*
Sample Input:

Enter your symptom: fever

Sample Output:

========================================
 Hospital Expert System
========================================

Available symptoms:
- chest pain
- cough
- fever
- headache
- stomach pain

Enter your symptom: fever

Possible Disease: Flu
Medical Advice: Take rest and drink plenty of fluids.
*/
#include <iostream>
#include <map>
#include <string>
#include <algorithm>
using namespace std;

// Function to convert string to lowercase
string toLowerCase(string str)
{
    transform(str.begin(), str.end(), str.begin(), ::tolower);
    return str;
}

int main()
{
    cout << "========================================" << endl;
    cout << " Hospital Expert System " << endl;
    cout << "========================================" << endl;

    // Knowledge base using map
    map<string, pair<string, string>> disease_data;

    disease_data["fever"] = {"Flu", "Take rest and drink plenty of fluids."};
    disease_data["cough"] = {"Cold", "Use cough syrup and stay hydrated."};
    disease_data["headache"] = {"Migraine", "Take proper sleep and avoid stress."};
    disease_data["stomach pain"] = {"Acidity", "Avoid oily food and drink more water."};
    disease_data["chest pain"] = {"Heart Problem", "Consult a doctor immediately."};

    cout << "\nAvailable symptoms:" << endl;

    for (auto item : disease_data)
    {
        cout << "- " << item.first << endl;
    }

    string user_symptom;

    cout << "\nEnter your symptom: ";
    getline(cin, user_symptom);

    user_symptom = toLowerCase(user_symptom);

    // Expert system logic
    if (disease_data.find(user_symptom) != disease_data.end())
    {
        cout << "\nPossible Disease: "
             << disease_data[user_symptom].first << endl;

        cout << "Medical Advice: "
             << disease_data[user_symptom].second << endl;
    }
    else
    {
        cout << "\nSorry! Symptom not found in database." << endl;
        cout << "Please consult a doctor." << endl;
    }

    return 0;
}